src/workspacer.Shared/Layout/FocusLayoutEngine.cs

```diff
@@ -78,6 +78,16 @@ public IEnumerable<IWindowLocation> CalcLayout(IEnumerable<IWindow> windows, int
             return list;
         }
 
+        private int GetLeftHeight(int spaceHeight, int numWindows, int numInPrimary)
+        {
+            return spaceHeight / Math.Max((numWindows - numInPrimary) / 2, 1);
+        }
+
+        private int GetRightHeight(int spaceHeight, int numWindows, int numInPrimary)
+        {
+            return spaceHeight / Math.Max((numWindows - numInPrimary + 1) / 2, 1);
+        }
```

Comment:
Looking at this code here, I'm trying to do some mental math to make sure I understand the formulas used.

Let's say we have:

* numWindows = 5
* numInPrimary = 2
* spaceHeight = 800

That gives us:

* windowsOutsidePrimary = 5 - 2 = 3
* numLeftWindows = Max((windowsOutsidePrimary/2), 1) = 1.5, except int, so no rounding = 1
* numRightWindows = Max(((windowsOutsidePrimary +1 )/2), 1) = 2

Which leaves us numLeftWindows + numRightWindows = 3. Which adds up to windowsOutsidePrimary. Great!

Now lets try with:

* numWindows = 6
* windowsOutsidePrimary = 6 - 2 = 4
* numLeftWindows = Max((windowsOutsidePrimary/2), 1) = 2
* numRightWindows = Max(((windowsOutsidePrimary +1 )/2), 1) = 2.5 except int, so no rounding = 2

And it still adds up. Right!

But I had to do all that math myself, and make up those test-cases, because there's no unit-tests for this bit right now.

Wouldn't it be nice to have tests for that, so that we know it works now and will keep working in the future? :)

Reply:
Thanks for all those suggestions. I’ve isolated the calculation of the number of windows for the 2 secondary panel. I’ve also added unit testing for those. And fixed some bugs and weird behavior